<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Richard Collins &lt;richard.collins@bristol.ac.uk&gt;</author>
 <comment>Copyright (C) University of Bristol 2017
   This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. 
   If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
   See LICENSE file for details.</comment>
 <class>DeviceDialog</class>
 <widget class="QDialog" name="DeviceDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>508</width>
    <height>330</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="4" column="4">
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <widget class="QLineEdit" name="dataPortAddress">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="dataPortPortLabel">
       <property name="text">
        <string>Port</string>
       </property>
       <property name="buddy">
        <cstring>dataPortPort</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="dataPortPort">
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>65535</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="5" column="1">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Net&amp;mask</string>
     </property>
     <property name="buddy">
      <cstring>deviceNetmask</cstring>
     </property>
    </widget>
   </item>
   <item row="5" column="4">
    <widget class="QLineEdit" name="deviceNetmask">
     <property name="inputMethodHints">
      <set>Qt::ImhUrlCharactersOnly</set>
     </property>
     <property name="inputMask">
      <string/>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QLabel" name="deviceAddressLabel">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Address/IP</string>
     </property>
     <property name="buddy">
      <cstring>dataPortAddress</cstring>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="deviceIcon">
     <property name="maximumSize">
      <size>
       <width>64</width>
       <height>64</height>
      </size>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="images.qrc">:/icons/DataPort</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="3" column="1" colspan="3">
    <widget class="QLabel" name="label_9">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>&amp;Network Type</string>
     </property>
     <property name="buddy">
      <cstring>dataPortType</cstring>
     </property>
    </widget>
   </item>
   <item row="3" column="4">
    <widget class="QComboBox" name="dataPortType">
     <property name="editable">
      <bool>true</bool>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string>tap</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>tun</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>eth</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>udp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>tcp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>tcpsrv</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="7" column="1" colspan="3">
    <widget class="QLabel" name="label_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Capture Le&amp;vel</string>
     </property>
     <property name="buddy">
      <cstring>dataPortCapture</cstring>
     </property>
    </widget>
   </item>
   <item row="8" column="4">
    <widget class="QFrame" name="bottomLayout">
     <layout class="QHBoxLayout" name="_2">
      <item>
       <widget class="QFrame" name="deviceConnectedTo">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2"/>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="9" column="2" colspan="3">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="1" column="1" colspan="3">
    <widget class="QLabel" name="label_8">
     <property name="text">
      <string>&amp;URI</string>
     </property>
     <property name="buddy">
      <cstring>deviceUri</cstring>
     </property>
    </widget>
   </item>
   <item row="7" column="4">
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <item>
      <widget class="QComboBox" name="dataPortCapture">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <item>
        <property name="text">
         <string>N/A</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>datagram</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>ip</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>eth</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="dataPortProm">
       <property name="text">
        <string>Promiscuous Mode</string>
       </property>
       <property name="tristate">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="4">
    <widget class="QLineEdit" name="deviceUri">
     <property name="font">
      <font>
       <pointsize>16</pointsize>
      </font>
     </property>
    </widget>
   </item>
   <item row="6" column="1">
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>Name</string>
     </property>
     <property name="buddy">
      <cstring>deviceName</cstring>
     </property>
    </widget>
   </item>
   <item row="6" column="4">
    <widget class="QLineEdit" name="deviceName">
     <property name="inputMethodHints">
      <set>Qt::ImhUrlCharactersOnly</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>deviceUri</tabstop>
  <tabstop>dataPortType</tabstop>
  <tabstop>dataPortAddress</tabstop>
  <tabstop>dataPortPort</tabstop>
  <tabstop>deviceNetmask</tabstop>
  <tabstop>deviceName</tabstop>
  <tabstop>dataPortCapture</tabstop>
  <tabstop>dataPortProm</tabstop>
 </tabstops>
 <resources>
  <include location="images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DeviceDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>408</x>
     <y>320</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DeviceDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>476</x>
     <y>320</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
